udfs/remote_udfs/nlp/call_nlp/main.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from google.cloud import language_v1
+from google.api_core.retry import Retry
+
+# https://cloud.google.com/functions/docs/bestpractices/tips#use_global_variables_to_reuse_objects_in_future_invocations
+client = language_v1.LanguageServiceClient()
+
+def remote_vertex_ai(request):
+    '''
+    This function serves the request.
+    The additional parameter "mode" is part of a key value pair sent as part of the call.
+    For more information on the UserDefinedContext see:
+    https://cloud.google.com/bigquery/docs/reference/standard-sql/remote-functions#input_format
+    '''
+    request_json = request.get_json()
+    mode = request_json['userDefinedContext']['mode']
+    # The input into the function from BigQuery is retrieved with the calls field. 
+    calls = request_json['calls']
+    if mode == "call_nlp":
+        return call_nlp(calls)
+
+    return json.dumps({"errorMessage": f"Not supported 'mode' in user_defined_context: {mode}"}), 400
+
+
+def call_nlp(calls):
+    try:
+        return_value = []
+        # Create the client to connecto the Language Service.
+        for call in calls:
+            # Retreive the text to be analyzed
+            text = call[0]
+            # Prepare the Document object to be sent to the service.
+            document = language_v1.Document(
+                content=text, type_=language_v1.Document.Type.PLAIN_TEXT
```

Comment:
pass value for type_ via the userDefinedContext

Reply:
I didn't pass in as a parameter but still used the if-statement style.